src/wallet/wallet.h

```diff
@@ -700,7 +700,16 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
 
     bool fFileBacked;
 
-    std::set<int64_t> setKeyPool;
```

Comment:
> holding a flexible data-structure (KeypoolCache) allows us to be more flexible in future

I don't have any problem with keeping `KeypoolCache` if it will be useful later.

What seems bad to me is using `std::set` with a custom comparison function to emulate a map. Would suggest removing `nIndex` and `operator<` from KeypoolCache and using `std::map<int64_t, KeypoolCache>`